src_c/key.c

```diff
@@ -804,13 +804,35 @@ key_set_text_input_rect(PyObject *self, PyObject *obj)
     /* https://wiki.libsdl.org/SDL_SetTextInputRect */
 #if IS_SDLv2
     SDL_Rect *rect, temp;
+    SDL_Window *sdlWindow = pg_GetDefaultWindow();
+    SDL_Renderer *sdlRenderer = SDL_GetRenderer(sdlWindow);
+
     if (obj == Py_None) {
         Py_RETURN_NONE;
     }
     rect = pgRect_FromObject(obj, &temp);
     if (!rect)
         return RAISE(PyExc_TypeError, "Invalid rect argument");
+
+    if (sdlRenderer!=NULL){
+        SDL_Rect vprect, rect2;
+	/* new rect so we do not overwrite the input rect */
+	float scalex, scaley;
```

Comment:
This PR is looking good! 👍 
Minor nitpick here, the indentation can be fixed for code cleanliness sake